src/Service.Tests/SqlTests/RestApiTests/Patch/PatchApiTestBase.cs

```diff
@@ -361,6 +361,51 @@ public virtual async Task PatchOne_Update_IfMatchHeaders_Test()
                 );
         }
 
+        /// <summary>
+        /// Test to validate successful execution of PATCH operation which satisfies the database policy for the operation (insert/update) it resolves into.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task PatchOneWithDatabasePolicy()
+        {
+            // PATCH operation resolves to update because we have a record present for given PK.
+            // Since the database policy for update operation ("@item.pieceid ne 1") is satisfied by the operation, it executes successfully.
+            string requestBody = @"
+            {
+                ""piecesAvailable"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "categoryid/100/pieceid/99",
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                    sqlQuery: GetQuery("PatchOneUpdateWithDatabasePolicy"),
+                    operationType: Config.Operation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK,
+                    clientRoleHeader: "database_policy_tester"
+                );
+
+            // PATCH operation resolves to insert because we don't have a record present for given PK.
+            // Since the database policy for insert operation ("@item.pieceid ne 6 and @item.piecesAvailable gt 0") is satisfied by the operation, it executes successfully.
+            requestBody = @"
+            {
+                ""piecesAvailable"": 4,
+                ""categoryName"": ""SciFi""
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "categoryid/0/pieceid/7",
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                    sqlQuery: GetQuery("PatchOneInsertWithDatabasePolicy"),
+                    operationType: Config.Operation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.Created,
+                    clientRoleHeader: "database_policy_tester",
+                    expectedLocationHeader: "categoryid/0/pieceid/7"
+                );
```

Comment:
Do you have a test for when this operation fails due to db policy? (if it's the other test you modified in this file, that's fine. I added a comment there for clarification.